cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java

```diff
@@ -121,17 +132,51 @@
 
   @Override
   protected void configure() {
-    bind(ArtifactRepositoryReader.class).toProvider(artifactRepositoryReaderProvider);
+    Boolean artifactLocalizerEnabled = cConf.getBoolean(Constants.Preview.ARTIFACT_LOCALIZER_ENABLED, false);
 
-    bind(ArtifactRepository.class).to(DefaultArtifactRepository.class);
-    expose(ArtifactRepository.class);
+    if (artifactLocalizerEnabled) {
+      // Use remote implementation to fetch artifact metadata from AppFab.
+      // Remote implementation internally uses artifact localizer to fetch and cache artifacts locally.
+      bind(ArtifactRepositoryReader.class).to(RemoteArtifactRepositoryReader.class);
+      bind(ArtifactRepository.class).to(RemoteArtifactRepository.class);
+      expose(ArtifactRepository.class);
+      bind(ArtifactRepository.class)
+        .annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO))
+        .to(RemoteArtifactRepository.class)
+        .in(Scopes.SINGLETON);
+      expose(ArtifactRepository.class).annotatedWith(Names.named(AppFabricServiceRuntimeModule.NOAUTH_ARTIFACT_REPO));
```

Comment:
nit: Do we need to expose annotated ArtifactRepository.class separately? In line 142, ArtifactRepository.class is already being exposed.

Reply:
I think we do need.